kubevirt/common-templates
# Add a script to check and download the latest fedora image and build …

**What this PR does / why we need it**:

This PR follows this process of identifying and creating a new Container Disk Image for Fedora and storing it in a known Quay registry
On a high level, the job should do the following:
1. Fetch the latest fedora disk image from fedora's webite
2. Fetch the latest fedora container-disk image from our quay image registry
3. Check if the image fetched in 1 is different than the qcow2 image contained in the quay image registry in 2, if they differ, download the latest Fedora Qcow2 image.  
4. Build a new Container Disk image with docker build, using the following commands  if not, exit with 0
Dockerfile contests.FROM scratch
COPY Fedora33.qcow2 /disk/                                                                                                                                                                                                                                                                                                                                                         
docker build . -t localhost:5000/mydisk
5. Push the image into the internal CI registry
docker push localhost:5000/mydisk
6. Run tests. The tests are run in the context of common-templates. So we build a VM from the OS template that the common-templates project provides which uses the newly created containerdisk image from the internal registry 
6. If a VM is created successfully using the new image. Push the new image into the Quay containerdisk image repository
 docker push quay.io/kubevirt/fedora-images:33.

The tasks are split into this other PR - https://github.com/kubevirt/project-infra/pull/1029

**Release note**:

```release-note
Script to check and download the latest fedora image and build  a CDI using it
```
Signed-off-by: Shweta Padubidri <spadubid@redhat.com>
